deepchem/splits/splitters.py

```diff
@@ -582,7 +583,10 @@ def split(self,
       np.random.seed(seed)
 
     mws = []
-    for smiles in dataset.ids:
+    # If the user did not pass in an array of SMILES strings, assume that dataset.ids contains SMILES strings
+    if not smiles_arr:
```

Comment:
It's better to write `if smiles_arr is None`.  There are lots of values other than `None` that can evaluate to False, so this creates a risk of bugs.

Reply:
Ok good call, I can make that change. Do you guys have any preferred way to add that commit to this pull request?

Reply:
Just add another commit to the origin branch (`amandajean119:master`) and it will automatically get added to the PR.

In the future, it's best to create a new branch in your repository to act as the origin for a PR instead of doing it from the master branch.

Reply:
Ok thanks for the info. This is my 2nd pull request ever :)